mqtt-simulator/topic.py

```diff
@@ -60,24 +70,45 @@ def generate_initial_value(self, data):
         elif data['TYPE'] == 'math_expression':
             self.expression_evaluators[data['NAME']] = ExpressionEvaluator(data['MATH_EXPRESSION'], data['INTERVAL_START'], data['INTERVAL_END'], data['MIN_DELTA'], data['MAX_DELTA'])
             return self.expression_evaluators[data['NAME']].get_current_expression_value()
+        elif data['TYPE'] == 'raw_values':
+            self.raw_values_index = data.get('INDEX_START', 0)
+            values = data['VALUES']
+            return values[self.raw_values_index]
             
     def generate_next_value(self, data, old_value):
-        if 'RESET_PROBABILITY' in data and random.random() < data['RESET_PROBABILITY']:
+        randN = random.random()
+        if randN < data.get('RESET_PROBABILITY', 0):
             return self.generate_initial_value(data)
-        if 'RESTART_ON_BOUNDARIES' in data and data['RESTART_ON_BOUNDARIES'] and (old_value == data['MIN_VALUE'] or old_value == data['MAX_VALUE']):
+        if data.get('RESTART_ON_BOUNDARIES', False) and (old_value == data.get('MIN_VALUE') or old_value == data.get('MAX_VALUE')):
             return self.generate_initial_value(data)
-        if random.random() < data['RETAIN_PROBABILITY']:
+        if randN < data.get('RETAIN_PROBABILITY', 0):
             return old_value
         if data['TYPE'] == 'bool':
             return not old_value
         elif data['TYPE'] == 'math_expression':
             return self.expression_evaluators[data['NAME']].evaluate_expression()
+        elif data['TYPE'] == 'raw_values':
+            values = data['VALUES']
+            # get boundaries or default to data length
+            startIndex = data.get('INDEX_START', 0)
+            endIndex = data.get('INDEX_END', len(values) - 1)
+            if endIndex >= len(values):
+                endIndex = len(values) - 1
+            # iterate then restart, return next, or disconnect on end of data
+            index = self.raw_values_index + 1
```

Comment:
I think it would be better if you increment `raw_values_index` directly and only use it as an index.

```suggestion
            self.raw_values_index += 1
```